tfhe/src/integer/gpu/server_key/radix/scalar_div_mod.rs

```diff
@@ -646,52 +773,81 @@ impl CudaServerKey {
         assert!(
             Scalar::BITS >= numerator_bits as usize,
             "The scalar divisor type must have a number of bits that is\
-            >= to the number of bits encrypted in the ciphertext"
+        >= to the number of bits encrypted in the ciphertext"
         );
 
-        // wrappings_abs returns Scalar::MIN when its input is Scalar::MIN (since in signed numbers
-        // Scalar::MIN's absolute value cannot be represented.
-        // However, casting Scalar::MIN to signed value will give the correct abs value
-        // If Scalar and Scalar::Unsigned have the same number of bits
-        let absolute_divisor = Scalar::Unsigned::cast_from(divisor.wrapping_abs());
-
-        let chosen_multiplier =
-            choose_multiplier(absolute_divisor, numerator_bits - 1, numerator_bits);
-
         let mut quotient: CudaSignedRadixCiphertext = numerator.duplicate_async(streams);
-
         let num_blocks = numerator.as_ref().d_blocks.lwe_ciphertext_count().0 as u32;
-
+        let msg_bits = self.message_modulus.0.ilog2() as usize;
         let lwe_dimension = self
             .key_switching_key
             .output_key_lwe_size()
             .to_lwe_dimension();
 
-        let is_absolute_divisor_one = absolute_divisor == Scalar::Unsigned::ONE;
-        let is_divisor_negative = divisor < Scalar::ZERO;
-        let l_exceed_threshold = chosen_multiplier.l >= numerator_bits;
-        let is_power_of_two =
-            absolute_divisor == (Scalar::Unsigned::ONE << chosen_multiplier.l as usize);
-        let multiplier_is_small = chosen_multiplier.multiplier
-            < (<Scalar::Unsigned as Reciprocable>::DoublePrecision::ONE << (numerator_bits - 1));
+        let mut scalar_properties = prepare_default_scalar_divisor();
```

Comment:
I would do all the prepare_default_scalar_divisor tasks in integer/gpu/mod.rs as for the other functions